### PR TITLE
Add architecture tests

### PR DESCRIPTION
**Pull request by kft**
## Description
This PR add a new test suite verification of architectural concerns.

### CODE REVIEW

File: source/B2B.Transactions.IntegrationTests/B2B.Transactions.IntegrationTests.csproj [hunk 1]
```diff
@@ -41,6 +41,7 @@ limitations under the License.
     </ItemGroup>
 
     <ItemGroup>
+      <ProjectReference Include="..\B2B.Transactions.Api\B2B.Transactions.Api.csproj" />
```
Comment (Sondergaard) [inline source/B2B.Transactions.IntegrationTests/B2B.Transactions.IntegrationTests.csproj:44]: Is this change needed? I original added this project reference to have access to Program-class in the API assembly. CompositionRootTest is moved to ArchitectureTest and this reference can safely be removed imo.

Comment (kft) [reply source/B2B.Transactions.IntegrationTests/B2B.Transactions.IntegrationTests.csproj:44]: No. You are right :-)

**Review by Sondergaard — APPROVED** [review]
